src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/CreateIndexAttempt.java

```diff
@@ -0,0 +1,143 @@
+package io.stargate.sgv2.jsonapi.service.operation.tables;
+
+import static io.stargate.sgv2.jsonapi.util.CqlIdentifierUtil.cqlIdentifierFromUserInput;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.type.DataType;
+import com.datastax.oss.driver.api.core.type.ListType;
+import com.datastax.oss.driver.api.core.type.MapType;
+import com.datastax.oss.driver.api.core.type.SetType;
+import com.datastax.oss.driver.api.querybuilder.SchemaBuilder;
+import com.datastax.oss.driver.api.querybuilder.schema.CreateIndex;
+import com.datastax.oss.driver.api.querybuilder.schema.CreateIndexOnTable;
+import com.datastax.oss.driver.api.querybuilder.schema.CreateIndexStart;
+import com.datastax.oss.driver.internal.querybuilder.schema.DefaultCreateIndex;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.cqldriver.override.ExtendedCreateIndex;
+import io.stargate.sgv2.jsonapi.service.operation.SchemaAttempt;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+ An attempt to create index for a table's column.
+*/
+public class CreateIndexAttempt extends SchemaAttempt<TableSchemaObject> {
+  private final String columnName;
+  private final String indexName;
+  private final TextIndexOptions textIndexOptions;
+  private final VectorIndexOptions vectorIndexOptions;
+  private final DataType dataType;
+  private final boolean ifNotExists;
+
+  /*
+   * @param position The position of the attempt in the sequence, for create index it's  always 0.
+   * @param schemaObject The schema object representing the table.
+   * @param columnName The name of the column to create the index on.
+   * @param dataType The data type of the column.
+   * @param indexName The name of the index to create.
+   * @param textIndexOptions The options for a text column type index.
+   * @param vectorIndexOptions The options for a vector column type index.
+   * @param ifNotExists Flag to ignore if index already exists.
+   * @return The attempt to create the index.
+   */
+  protected CreateIndexAttempt(
+      int position,
+      TableSchemaObject schemaObject,
+      String columnName,
+      DataType dataType,
+      String indexName,
+      TextIndexOptions textIndexOptions,
+      VectorIndexOptions vectorIndexOptions,
+      boolean ifNotExists) {
+    super(position, schemaObject, new SchemaRetryPolicy(2, Duration.ofMillis(10)));
```

Comment:
Is the idea that we'll always retry? Even with `ifNotExists` being `false`?

Reply:
TBH, we didn't spec `ifNotExists` but did it since we have support in query builder. If it's creating confusion should I remove this feature?

Reply:
May be worth discussing: we should probably always enable "IF NOT EXISTS" for idempotent operation.

But since we already have that for CreateTableCommand, maybe it makes sense to expose it.

Reply:
That's a good catch, not sure what it means to retry with if not exists is false. In other words what to do on timeouts.

Reply:
Exactly.